src/core/work_queue.cpp

```diff
@@ -0,0 +1,37 @@
+#include "work_queue.hpp"
+
```

Comment:
Don't worry about the implementation of the work_queue thread pool class, just get familiar with the interface in the corresponding header.